File: src/main/scala/Foobar.scala
object Foobar {
  def main(args: Array[String]): Unit = {
    val n = 100
    foobar(n)
  }

  private def foobar(n: Int): Unit = {
    for (i <- 1 to n) {
      val divisibleBy3 = i % 3 == 0
      val divisibleBy5 = i % 5 == 0

      if (divisibleBy3 && divisibleBy5) {
        println("Foobar")
      } else if (divisibleBy3) {
        println("Foo")
      } else if (divisibleBy5) {
        println("Bar")
      } else {
        println(i)
      }
    }
  }
}

File: src/main/python/foobar.py
def foobar(n):
    for i in range(1, n + 1):
        divisible_by_3 = (i % 3 == 0)
        divisible_by_5 = (i % 5 == 0)

        if divisible_by_3 and divisible_by_5:
            print("Foobar")
        elif divisible_by_3:
            print("Foo")
        elif divisible_by_5:
            print("Bar")
        else:
            print(i)

if __name__ == "__main__":
    ns = 100
    foobar(ns)
